logical/plugin/system.go

```diff
@@ -0,0 +1,273 @@
+	if err != nil {
+		return consts.ReplicationDisabled
+	}
+
+	return reply.ReplicationState
+}
+
+func (s SystemViewClient) ResponseWrapData(data map[string]interface{}, ttl time.Duration, jwt bool) (*wrapping.ResponseWrapInfo, error) {
+	var reply ResponseWrapDataReply
+	args := &ResponseWrapDataArgs{
+		Data: data,
+		TTL:  ttl,
+		JWT:  jwt,
+	}
+
+	err := s.client.Call("Plugin.ResponseWrapData", args, &reply)
+	if err != nil {
+		return nil, err
+	}
+	if reply.Error != nil {
+		return nil, reply.Error
+	}
+
+	return reply.ResponseWrapInfo, nil
+}
+
+func (s SystemViewClient) LookupPlugin(name string) (*pluginutil.PluginRunner, error) {
+	var reply LookupPluginReply
+
+	err := s.client.Call("Plugin.LookupPlugin", name, &reply)
+	if err != nil {
+		return nil, err
+	}
+	if reply.Error != nil {
+		return nil, reply.Error
+	}
+
+	return reply.PluginRunner, nil
+}
+
+func (s SystemViewClient) MlockEnabled() bool {
+	var reply MlockEnabledReply
+	err := s.client.Call("Plugin.MlockEnabledReply", new(interface{}), &reply)
+	if err != nil {
+		return false
+	}
+
+	return reply.MlockEnabled
+}
+
+type SystemViewServer struct {
+	impl logical.SystemView
+}
+
+func (s SystemViewServer) DefaultLeaseTTL(_ interface{}, reply *DefaultLeaseTTLReply) error {
+	ttl := s.impl.DefaultLeaseTTL()
+	*reply = DefaultLeaseTTLReply{
+		DefaultLeaseTTL: ttl,
+	}
+
+	return nil
+}
+
+func (s SystemViewServer) MaxLeaseTTL(_ interface{}, reply *MaxLeaseTTLReply) error {
+	ttl := s.impl.MaxLeaseTTL()
+	*reply = MaxLeaseTTLReply{
+		MaxLeaseTTL: ttl,
+	}
+
+	return nil
+}
+
+func (s SystemViewServer) SudoPrivilege(args *SudoPrivilegeArgs, reply *SudoPrivilegeReply) error {
+	sudo := s.impl.SudoPrivilege(args.Path, args.Token)
+	*reply = SudoPrivilegeReply{
+		Sudo: sudo,
+	}
+
+	return nil
+}
+
+func (s SystemViewServer) Tainted(_ interface{}, reply *TaintedReply) error {
+	tainted := s.impl.Tainted()
+	*reply = TaintedReply{
+		Tainted: tainted,
+	}
+
+	return nil
+}
+
+func (s SystemViewServer) CachingDisabled(_ interface{}, reply *CachingDisabledReply) error {
+	cachingDisabled := s.impl.CachingDisabled()
+	*reply = CachingDisabledReply{
+		CachingDisabled: cachingDisabled,
+	}
+
+	return nil
+}
+
+func (s SystemViewServer) ReplicationState(_ interface{}, reply *ReplicationStateReply) error {
+	replicationState := s.impl.ReplicationState()
+	*reply = ReplicationStateReply{
+		ReplicationState: replicationState,
+	}
+
+	return nil
+}
+
+func (s SystemViewServer) ResponseWrapData(args *ResponseWrapDataArgs, reply *ResponseWrapDataReply) error {
+	info, err := s.impl.ResponseWrapData(args.Data, args.TTL, args.JWT)
+	if err != nil {
+		*reply = ResponseWrapDataReply{
+			Error: plugin.NewBasicError(err),
+		}
+		return nil
+	}
+	*reply = ResponseWrapDataReply{
+		ResponseWrapInfo: info,
+	}
+
+	return nil
+}
+
+func (s SystemViewServer) LookupPlugin(args string, reply *LookupPluginReply) error {
+	runner, err := s.impl.LookupPlugin(args)
```

Comment:
Should we allow chaining of plugins? I'm not sure we would want a backend plugin to spawn more plugins. What do you think?